### PR TITLE
Vision stack implementation

### PR DESCRIPTION
**Pull request by DaniParr**
## Description

Added the vision_stack package that will serve as the base for our vision pipeline for everything cv and ml!

## Related Issues

\- Closes #XXX

## About This PR

- [ ] I have updated documentation related to this change so that future members are aware of the changes I've made.

### CODE REVIEW

File: SubjuGator/perception/subjugator_perception/nodes/vision_stack_test.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+__author__ = "Daniel Parra"
+
+
+class ObjectDetectionTest:
+    def __init__(self):
+        camera = rospy.get_param("~image_topic", "/camera/front/right/image_rect_color")
+        SIZE = (960, 608)
+        self.vs = VisionStack(
+            layers=[
+                ResizeLayer((0, 0), 960, 608),
```
Comment (cbrxyz) [inline SubjuGator/perception/subjugator_perception/nodes/vision_stack_test.py:18]: This should be
```python
ResizeLayer((0, 0), SIZE[0], SIZE[1])
```
right?

**Review by cbrxyz — CHANGES_REQUESTED** [review]
Thanks! Just one small question.

File: SubjuGator/perception/subjugator_perception/nodes/vision_stack_test.py (hunk 1 above)
Comment (DaniParr) [reply SubjuGator/perception/subjugator_perception/nodes/vision_stack_test.py:18]: The first argument is the initial size, and the next parameters are the new width and new heights. However, the initial size of the image is not being used in any of the layers...

Comment (DaniParr) [reply SubjuGator/perception/subjugator_perception/nodes/vision_stack_test.py:18]: Ill change that rn

**Review by cbrxyz — DISMISSED** [review]
Looks great!

**Pull request merged**